src/modules/users/__tests__/users.controller.spec.ts

```diff
@@ -52,161 +53,313 @@ describe('modules/users/UsersController', () => {
-    };
-    const params = { id: '1234' };
-
-    usersService.findByAuth0Id = jest.fn(() => Promise.resolve(<User>{ _id: new ObjectIdMock(request.user.auth0Id), roles: [] }));
-    usersService.findById = jest.fn(() => Promise.resolve(<User>{ _id: new ObjectIdMock(params.id) }));
-    usersService.remove = jest.fn(() => Promise.resolve({ ok: 1 }));
-    mentorsService.removeAllApplicationsByUserId = jest.fn(() => Promise.resolve());
-    auth0Service.getAdminAccessToken = jest.fn(() => Promise.resolve({}));
-    auth0Service.deleteUser = jest.fn(() => Promise.resolve());
-
-    await usersController.remove(request, params)
-
-    expect(usersService.remove).toHaveBeenCalled();
-    expect(usersService.remove).toHaveBeenCalledWith(params.id);
-  });
-
-  it('should remove user from auth0', async () => {
-    const request = {
-      user: { auth0Id: '1234' },
-    };
-    const params = { id: '1234' };
-
-    usersService.findByAuth0Id = jest.fn(() => Promise.resolve(<User>{ _id: new ObjectIdMock(request.user.auth0Id), roles: [] }));
-    usersService.findById = jest.fn(() => Promise.resolve(<User>{ _id: new ObjectIdMock(params.id), auth0Id: 'abcd' }));
-    usersService.remove = jest.fn(() => Promise.resolve({ ok: 1 }));
-    mentorsService.removeAllApplicationsByUserId = jest.fn(() => Promise.resolve());
-    auth0Service.getAdminAccessToken = jest.fn(() => Promise.resolve({ access_token: '159' }));
-    auth0Service.deleteUser = jest.fn(() => Promise.resolve());
-
-    await usersController.remove(request, params)
-
-    expect(auth0Service.getAdminAccessToken).toHaveBeenCalled();
-    expect(auth0Service.deleteUser).toHaveBeenCalled();
-    expect(auth0Service.deleteUser).toHaveBeenCalledWith('159', 'abcd');
+  
+  describe('currentUser', () => {
+    it('should return the current user', async () => {
+      const request = { user: { auth0Id: '123' } };
+      const data: User = <User>({ _id: 123, name: 'Crysfel Villa'});
+      const response = { success: true, data };
+      usersService.findByAuth0Id = jest.fn(() => Promise.resolve(data));
+
+      expect(await usersController.currentUser(request)).toEqual(response);
+    });
+
+    it('should create a new user', async () => {
+      const request = { user: { auth0Id: '123' } };
+      const data: User = <User>({ _id: '123', name: 'Crysfel Villa' });
+      const response = { success: true, data };
+
+      usersService.findByAuth0Id = jest.fn(() => Promise.resolve(undefined));
+      usersService.findByEmail = jest.fn(() => Promise.resolve(undefined));
+      usersService.create = jest.fn(() => Promise.resolve(data));
+      emailService.send = jest.fn();
+      auth0Service.getAdminAccessToken = jest.fn(() => Promise.resolve({ access_token: 'abc' }));
+      auth0Service.getUserProfile = jest.fn(() => Promise.resolve({ _id: '123' }));
+
+      expect(await usersController.currentUser(request)).toEqual(response);
+      expect(usersService.create).toHaveBeenCalled();
```

Comment:
What do think about asserting that the method was called only once?

```suggestion
      expect(usersService.create).toHaveBeenCalledTimes(1);
```

Reply:
Done!